fileServer.js

```diff
@@ -10,31 +10,33 @@ const mimeType = {
   'jpg': 'image/jpeg',
   'png': 'image/png',
   'js': 'text/javascript',
-  'css': 'text/css'
+  'css': 'text/css',
+  'svg': 'image/svg+xml',
+  'ico': 'image/x-icon',
+  'gif': 'image/gif',
 };
 
-function generate500error(response, errorMessage) {
-
-  Error.captureStackTrace(this, generate500error);
+function generate500error(response) {
   var err = new Error();
-  response.writeHead(500, {
-    'Content-Type': 'text/html'
-  });
+  fs.readFile('public/500.html', 'UTF-8', function(error, html) {
+    if (error) {
+      throw err;
+    }
+    response.writeHead(500, {
+      'Content-Type': 'text/html'
+    });
+    console.log(`Server cannot process the request: 500 internal error`);
```

Comment:
Much better logic - to make it easy for an operations person it is generally preferable to have the message + stack trace all on one line - so they can use tools like grep, awk, ack, etc to pull lines out of the files easily. 

Today we will look at some errors from the systems my team runs and the logs they produce to expose you to some of what i mean in practice.